### ISSUE TITLE
Should we add history variable dimensions for land-use-category and land-use-category x pft?

### ISSUE THREAD

**Issue by ckoven** [NGEET/fates#964]
We currently have a few history variables that are specific to secondary forest, such as these:
https://github.com/NGEET/fates/blob/main/main/FatesHistoryInterfaceMod.F90#L5318-L5351

The new PR https://github.com/NGEET/fates/pull/888 adds considerably to that list of variables that apply to secondary lands only, see the new ones here:
https://github.com/NGEET/fates/pull/888/files#diff-03feca0f1df8d3791c0d8f7b6055a0368d5241716d19d6a97c3d227fd6c5ad23R4926-R6904

This seems like something that may start to become a bit unmanageable as we add new land use types in FATES.  So I wanted to suggest that we add a new history variable dimension for these land use categories, which we can then index as 1 = primary lands, 2 = secondary, etc.  That way we can consolidate some of these variables into that dimension and simplify things.  

If we do this, the question is as always where to stop. I can imagine that it would be very useful to add a land-use by PFT dimension too, that way we can keep track of how different PFTs exist and behave differently on different land use categories.  Do we also add land-use x age? etc.  I'm not sure.  I guess my initial instinct would be to just add land-use and land-use x pft and stop there.  That would leave some variables still as separate secondary-forest-only variables (like these two: https://github.com/NGEET/fates/blob/main/main/FatesHistoryInterfaceMod.F90#L5339-L5351), which I think is ok as I think the content of those is pretty specific to secondary lands and wouldn't apply to, e.g., pasture. Likewise some of the new ones from #888. But interested to hear if others have thoughts.

I propose to do this after #888 is merged to main.

**Comment by sshu88** [comment]
@ckoven Thank you for the suggestion. It will be very helpful to have different land-use categories as an additional dimension. I'm thinking more about how we form a reasonable list of this "land-use-category" to be applicable for the longer term when we have more managed patches from different types of managements. How about using disturbance type (harvested, LUC, fire, irrigated, ...) instead of the simple primary vs. secondary tag list. In this case secondary forest will be those patches with disturbance dimension of harvested and LUC.

**Comment by ckoven** [comment]
HI @sshu88 -- right, the idea would be that as we build out more land use categories, we would add to the length of that dimension. Hopefully we can do it in a fairly automatic way so that we don't have to increment anything manually in HLMs as we add a new category.

**Comment by ckoven** [comment]
@sshu88 and all, just to be more specific, the idea would be to tie this new dimension directly to the indices we are currently using here: https://github.com/NGEET/fates/blob/main/main/FatesConstantsMod.F90#L34-L36

**Comment by rgknox** [comment]
@ckoven, I don't forsee any issues in extending out the new dimension as we add new land categories.
